### ISSUE TITLE
Restart does not restore windows to their original spaces

### ISSUE THREAD

**Issue by tplobo** [microsoft/vscode#224776]
Type: <b>Bug</b>

I usually work with multiple projects, each with its own VSCodes window. When I restart VSCode to update it, the multiple windows are re-opened in the current space, instead of the spaces they originally were. This creates unnecessary effort to re-organize my windows in each space.

VS Code version: Code 1.92.0 (Universal) (b1c0a14de1414fcdaa400695b4db1c0799bc3124, 2024-07-31T23:26:45.634Z)
OS version: Darwin arm64 23.6.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M3 Max (16 x 2400)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|3, 5, 5|
|Memory (System)|64.00GB (2.44GB free)|
|Process Argv|--crash-reporter-id d49e38ce-4860-40cf-8059-5d65c06d0a0f|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (31)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-sqlite|ale|0.14.1
aws-toolkit-vscode|ama|3.18.0
preview-pdf|ana|1.0.0
preview-tiff|ana|1.0.1
solargraph|cas|0.24.1
ruff|cha|2024.36.0
json-tree-view|Cha|3.6.1
gitlens|eam|15.2.3
vscode-great-icons|emm|2.1.105
prettier-vscode|esb|10.4.0
pythonsnippets|frh|1.0.2
copilot|Git|1.219.0
copilot-chat|Git|0.18.1
vscode-pull-request-github|Git|0.92.0
restore-git-branch-tabs|gko|0.4.1
applescript|idl|0.25.1
vscode-plist|ivh|0.1.3
debugpy|ms-|2024.10.0
python|ms-|2024.13.2024080201
jupyter|ms-|2024.7.0
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.0.19
vscode-jupyter-cell-tags|ms-|0.1.9
vscode-jupyter-slideshow|ms-|0.1.6
macos-native-copy-paste|qt-|1.2.2
sqlite-viewer|qwt|0.5.11
ruby-lsp|Sho|0.7.14
lua|sum|3.10.1
vscode-counter|uct|3.5.0
vscode-todo-highlight|way|1.0.5
vscode-ruby|win|0.28.0

(2 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscoreces:30445986
vscod805:30301674
binariesv615:30325510
vsaa593:30376534
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
962ge761:30959799
pythongtdpath:30769146
welcomedialogc:30910334
pythonnoceb:30805159
asynctok:30898717
pythonregdiag2:30936856
pythonmypyd1:30879173
2e7ec940:31000449
pythontbext0:30879054
accentitlementsc:30995553
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
01bff139:31013167
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
dwnewjupytercf:31046870
2f103344:31071589
impr_priority:31102340
refactort:31108082
ccplt:31103426
pythonrstrctxt:31103193
wkspc-onlycs-c:31106320

```

</details>

**Comment by tplobo** [comment]
Oh, I am sorry @bpasero, I did not use the right keywords when searching through past issues. Anyhow, I think the original issue of which this one is a duplicate is #979 (the `duplicate` tag is not clear). There has been a lot of discussion there since it was closed, and I'd like to know if this could not be reconsidered. I currently use a [custom solution](https://github.com/tplobo/restore-spaces) for my own spaces organization, but this seems a bit of an overkill for a problem that happens multiple times per month. Thanks in any case.
